akka-http-tests/src/test/scala/akka/http/scaladsl/server/TestServer.scala

```diff
@@ -42,19 +63,44 @@ object TestServer extends App {
           complete(index)
         }
       } ~
-        path("secure") {
-          authenticateBasicPF("My very secure site", auth) { user ⇒
-            complete(<html><body>Hello <b>{ user }</b>. Access has been granted!</body></html>)
-          }
-        } ~
-        path("ping") {
-          complete("PONG!")
+      path("secure") {
+        authenticateBasicPF("My very secure site", auth) { user ⇒
+          complete(<html>
+            <body>Hello
+              <b>
+                {user}
+              </b>
+              . Access has been granted!</body>
+          </html>)
+        }
+      } ~
+      path("ping") {
+        complete("PONG!")
+      } ~
+      path("crash") {
+        complete(sys.error("BOOM!"))
+      } ~
+      path("tweet") {
+        complete(Tweet("Hello, world!"))
+      } ~
+      (path("tweets") & parameter('n.as[Int])) { n => 
+        get {
+          val tweets = Source.repeat(Tweet("Hello, world!")).take(n)
+          complete(ToResponseMarshallable(tweets))
         } ~
-        path("crash") {
-          complete(sys.error("BOOM!"))
+        post {
+          entity(asSourceOf[Tweet]) { tweets ⇒
```

Comment:
Heh, one downside is that one would have to say 

```
        entity(as[Source[Measurement, _]]) { measurements =>
```

or NotUsed there. I'm looking into making it work anyway